### ISSUE TITLE
Load balancing: let the backends indicate the weight/load

### ISSUE THREAD

**Issue by gaetancollaud** [containous/traefik#3745]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?
Currently for the load-balancing we can use the `wrr` balancer which is a bit static and `drr` balancer which will "increases weights on servers that perform better than others" ([doc quote](https://docs.traefik.io/basics/#load-balancing)). 

Could we have a way that the backends announce their weight or load dynamically ?

Not sure if we need to adapt the `drr` to fit this request or if we need a complete new load balancing system.

#### Use case 1: Websocket
I'm using a lot websocket connections. Those connections should be load balanced over some containers. But since it's not small calls that end quickly I can't really use the `drr`. I already monitor the number of connections per backend. Traefik could easily request the number of connections or the "load" of each backend using a REST call and redirect new requests to the node with the less connections.

#### Use case 2: Uneven response time
I assume the `drr` look at response time. But some requests are longer than other in an application. Maybe it's not related to load, it's just due to some waiting time. It's not really fair to base the weight on the response time. It would be better if each node can tell what its real load is.

**Comment by davidjaquier** [comment]
Agree, it could be great to have something like the agent-check feature introduced in 1.5 version of HAProxy.

**Comment by alexeckert** [comment]
I like traefik and would like to use it, however its lack of supported lb algos currently prevent me from using it. I'd also like some connection based lb logic, e.g. like `least_conn` in nginx or HAProxy: http://nginx.org/en/docs/http/ngx_http_upstream_module.html#least_conn
Furthermore one could add a more detailed description what the `drr` option actually does.

**Comment by ishowta** [comment]
For context, the Power of Two Choices algorithm (similar to Least Connections) was added recently:
https://github.com/traefik/traefik/pull/11547

There have also been some mentions of Least Connections:
https://github.com/traefik/traefik/issues/4380
https://github.com/traefik/traefik/pull/10534#issue-2195663286
https://github.com/traefik/traefik/issues/6985#issuecomment-769240270
